### PR TITLE
Ensure that constant propagation patterns that rely on types having static shapes are only run if the types have a static shape

### PR DESCRIPTION
**Pull request by jorickert**
Otherwise these constprop patterns will crash

I plan to upstream this

### CODE REVIEW

File: src/Dialect/ONNX/Transforms/ConstProp.cpp [hunk 1]
```diff
@@ -1014,8 +1016,9 @@ Value ConstPropConcat(PatternRewriter &rewriter, Value replacingValue,
     ValueRange operands, IntegerAttr axisAttr) {
   ShapedType outputType = mlir::cast<ShapedType>(replacingValue.getType());
   int64_t axis = axisAttr.getValue().getSExtValue();
-  if (axis < 0)
+  if (axis < 0) {
     axis += outputType.getRank();
+  }
```
Comment (flemairen6) [inline src/Dialect/ONNX/Transforms/ConstProp.cpp]: Unnecessary change

**Review by flemairen6 — APPROVED** [review]

**Review by ttjost — APPROVED** [review]
